Add badge to thread screen

larkox opened the pull request:
#### Summary
Add mention count to Global Threads and Thread screens.

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-50034

#### Release Note
```release-note
Global Threads and Thread screens now show a badge if you have mentions on other threads / channels.
```

enahum commented on app/screens/thread/thread_header/thread_header.tsx:
```diff
@@ -0,0 +1,77 @@
+
+    const title = intl.formatMessage({id: 'thread.header.thread', defaultMessage: 'Thread'});
+    const channelName = displayName || intl.formatMessage({id: 'thread.header.unknownChannel', defaultMessage: 'Unknown'});
+    const subtitle = intl.formatMessage({id: 'thread.header.thread_in', defaultMessage: 'in {channelName}'}, {channelName});
```
the `in` prefix does not apply to DM's

enahum reviewed (changes requested):
a couple of nits, the rest LGTM

abhijit-singh commented:
@larkox I'd vote to remove the count from the single `thread` screen. It seems incorrect to show that number two levels down when clicking on it will take you back to the channel/global threads view and not the home screen where that number is coming from. Also checked a few other apps that have a similar pattern and hide the number when the user is more than 1 level down. Let's have a discussion in the team sync if you feel strongly we should have it there.

larkox commented:
Reverted the changes and updated the description. @abhijit-singh Feel free to test again (but I don't expect much changes apart of the mention not being visible now on the thread screen).

mattermost-build commented:
Building app in separate branch.

abhijit-singh reviewed (approved):
Looks good! 

Unrelated to this PR directly and very minor, but noticed that the vertical alignment of the number inside the badge is off. It should be moved down by a point. 
<img width="700" alt="image" src="https://github.com/mattermost/mattermost-mobile/assets/14369607/28411c95-4694-4c8a-84f2-454a92c4386d">

mattermost-build commented:
Building app in separate branch.
